commit 06bf6559d5f515323f8819b19d04ea932c2dc6aa
Author: Hugo Pereira Da Costa <hugo@oxygen-icons.org>
Date:   Mon Jan 3 11:41:28 2011 +0100

    Use GtkOrientable interface to get slider orientation for GtkScale widgets.
    CCBUG: 261932

diff --git a/src/oxygenstylewrapper.cpp b/src/oxygenstylewrapper.cpp
index c94241d..229cb4e 100644
--- a/src/oxygenstylewrapper.cpp
+++ b/src/oxygenstylewrapper.cpp
@@ -812,21 +812,26 @@ namespace Oxygen
             StyleOptions options = StyleOptions( widget, state, shadow );
             Style::instance().renderMenuItemRect( window, clipRect, widget, x, y, w, h, options );
 
-        } else if( d.isTroughAny() && GTK_IS_VSCALE( widget ) ) {
+        } else if( d.isTroughAny() && GTK_IS_SCALE( widget ) ) {
 
-            // TODO: calculate this value from the style "slider-width" property
+            const bool vertical( gtk_orientable_get_orientation( GTK_ORIENTABLE( widget ) ) == GTK_ORIENTATION_VERTICAL );
             const int offset( 6 );
-            if( d.isTrough() ) Style::instance().renderSliderGroove( window, clipRect, x, y + offset, w, h - 2*offset, Vertical );
-            else if( d.isTroughLower() ) Style::instance().renderSliderGroove( window, clipRect, x, y + offset, w, h, Vertical );
-            else if( d.isTroughUpper() ) Style::instance().renderSliderGroove( window, clipRect, x, y, w, h - offset, Vertical );
+            if( vertical ) {
 
-        } else if( d.isTroughAny() && GTK_IS_HSCALE( widget ) ) {
+                // TODO: calculate this value from the style "slider-width" property
+                if( d.isTrough() ) Style::instance().renderSliderGroove( window, clipRect, x, y + offset, w, h - 2*offset, Vertical );
+                else if( d.isTroughLower() ) Style::instance().renderSliderGroove( window, clipRect, x, y + offset, w, h, Vertical );
+                else if( d.isTroughUpper() ) Style::instance().renderSliderGroove( window, clipRect, x, y, w, h - offset, Vertical );
 
-            // TODO: calculate this value from the style "slider-width" property
-            const int offset( 6 );
-            if( d.isTrough() ) Style::instance().renderSliderGroove( window, clipRect, x + offset, y, w - 2*offset, h, StyleOptions() );
-            else if( d.isTroughLower() ) Style::instance().renderSliderGroove( window, clipRect, x + offset, y, w, h, StyleOptions() );
-            else if( d.isTroughUpper() ) Style::instance().renderSliderGroove( window, clipRect, x, y, w - offset, h, StyleOptions() );
+            } else {
+
+                // TODO: calculate this value from the style "slider-width" property
+                const int offset( 6 );
+                if( d.isTrough() ) Style::instance().renderSliderGroove( window, clipRect, x + offset, y, w - 2*offset, h, StyleOptions() );
+                else if( d.isTroughLower() ) Style::instance().renderSliderGroove( window, clipRect, x + offset, y, w, h, StyleOptions() );
+                else if( d.isTroughUpper() ) Style::instance().renderSliderGroove( window, clipRect, x, y, w - offset, h, StyleOptions() );
+
+            }
 
         } else if( d.isTrough() && shadow == GTK_SHADOW_IN ) {
 
